### ISSUE TITLE
Tooltip and Popover interference from overlay scrollbars

### ISSUE THREAD

**Issue by r20** [ant-design/ant-design#37709]
### Reproduction link

[https://ant.design/components/popover/](https://ant.design/components/popover/)

### Steps to reproduce

Open the Popover example at https://ant.design/components/popover/
Using Chrome debugger add these styles to the h2 element with text "Examples": overflow-x: overlay;
    width: 100px; (This will create an overlay scrollbar on the word Examples)
Hover over the "Hover me" button below "Examples".
Move the mouse up to be within the popover.
When the mouse goes over the scrollbar that's underneath the popover, the popover will close.
I was NOT able to reproduce it in a plain sandbox.  It's in my code, and in the antd popover documentation code.

### What is expected?

The popover stays open when hovering within the popover.

### What is actually happening?

Scrollbars created via overlay that are underneath popovers and tooltips, interfere with them causing them to close.  Regular scrollbars do not, only those that are overflow: overlay.  And, this only happens in some usages, such as at the popover documentation page.

| Environment | Info |
| --- | --- |
| antd | 4.23.2 |
| React | 16.14.0 |
| System | I don't know what this means |
| Browser | Chrome, Edge |

---
Other tooltip and popover libraries I have tried do not have this problem.  Although the problem can be reproduced at the popover and tooltip example pages, I do NOT see the problem with a plain reproduction, like here: https://codesandbox.io/s/basic-antd-4-23-2-forked-wmqbse?file=/demo.js:119-194.  Also, I can't reproduce the problem with using just rc-tooltip.

I had the problem in my code (overlay scrollbar interfering with open popovers and tooltips), and I see the same thing happens at https://ant.design/components/popover/.  I can't figure out the difference between what's on https://ant.design/components/popover/ and https://codesandbox.io/s/basic-antd-4-23-2-forked-wmqbse?file=/demo.js:119-194 that would cause this. 

This also happens in antd version 3.

**Comment by afc163** [comment]
Could you record a video?

**Comment by r20** [comment]
Here's a video: https://www.screenpresso.com/=eSeFb
With Chrome debugger I create a scrollbar first on "Examples".  Since it's underneath the popover, when I hover within the popover and over the scrollbar it closes.

**Comment by afc163** [comment]
I can't reproduce in Mac Chrome. Is it a browser or window-only bug?

<img width="256" alt="图片" src="https://user-images.githubusercontent.com/507615/191910636-5bd5d9cc-21b1-48fd-8aa8-2912bfe7a825.png">

**Comment by r20** [comment]
Did you use overflow: overlay?  (Because it does not happen with overflow: auto).

And, as I mentioned originally, I do NOT see the problem with a plain reproduction, like here: https://codesandbox.io/s/basic-antd-4-23-2-forked-wmqbse?file=/demo.js:119-194.  There must be something different about the documentation page.

**Comment by r20** [comment]
Also, I did mention I saw it in both Chrome and Edge.  Firefox doesn't support overflow: overlay.

**Comment by afc163** [comment]
Yes, I reproduce in my Chrome now! `overflow: overlay` is a deprecated usage, maybe you should drop it?

<img width="348" alt="图片" src="https://user-images.githubusercontent.com/507615/192095845-f0d412ae-31d1-4319-a8f8-9b88e9438865.png">

**Comment by r20** [comment]
Overlay is the only way to achieve what we want. 

 Do you know of anything different between the popover documentation page at https://ant.design/components/popover/ and what's on https://codesandbox.io/s/basic-antd-4-23-2-forked-wmqbse?file=/demo.js:119-194 (where it does not happen) ?

**Comment by afc163** [comment]
I don't know why but when I remove `span.all-code-box-controls` from dom tree it never happen again.

<img width="418" alt="图片" src="https://user-images.githubusercontent.com/507615/192106612-44344245-fc36-404f-823f-39aaf40a1ffa.png">

**Comment by r20** [comment]
@afc163 Interesting.  From your finding, I think I've narrowed it down to it happening if "position: relative" is used anywhere within the element having an overlay scrollbar.  I can reproduce it in a more simple environment now.  https://codesandbox.io/s/basic-antd-4-23-2-forked-8oskmx?file=/demo.js

I don't know that there's anything that can be done about it, but that's good to know.

**Comment by r20** [comment]
I don't think there's anything that can be done.
